armi/reactor/reactors.py

```diff
@@ -1679,68 +1747,102 @@ def findNeighbors(
         """
         Find assemblies that are next to this assembly.
 
-        Return a list of neighboring assemblies from the 30 degree point (point 1) then
-        counterclockwise around.
+        Return a list of neighboring assemblies.
+
+        For a hexagonal grid, the list begins from the 30 degree point (point 1)
+        then moves counterclockwise around.
+
+        For a Cartesian grid, the order of the neighbors is east, north, west,
+        south.
 
         .. impl:: Retrieve neighboring assemblies of a given assembly.
             :id: I_ARMI_R_FIND_NEIGHBORS
             :implements: R_ARMI_R_FIND_NEIGHBORS
 
+            This method takes an :py:class:`Assembly
+            <armi.reactor.assemblies.Assembly` as an input parameter and returns
+            a list of the assemblies neighboring that assembly. There are 6
+            neighbors in a hexagonal grid and 4 neighbors in a Cartesian grid.
+            The (i, j) indices of the neighbors are provided by
+            :py:meth:`getNeighboringCellIndices
+            <armi.reactor.grids.StructuredGrid.getNeighboringCellIndices>`. For
+            a hexagonal grid, the (i, j) indices are converted to (ring,
+            position) indexing using the core.spatialGrid instance attribute.
+
+            The `showBlanks` option determines whether non-existing assemblies
+            will be indicated with a `None` in the list or just excluded from
+            the list altogether.
+
+            The `duplicateAssembliesOnReflectiveBoundary` setting only works for
+            1/3 core symmetry with periodic boundary conditions. For these types
+            of geometries, if this setting is `True`, neighbor lists for
+            assemblies along a periodic boundary will include the assemblies
+            along the opposite periodic boundary that are effectively neighbors.
+
```

Comment:
Did we use 1/3 core symmetry with Cartesian grid?  No test for it.

Reply:
There is an input validator that checks the combinations of symmetry and geometry type, which will prevent a user from requesting 1/3 core symmetry with Cartesian grid.

https://github.com/terrapower/armi/blob/14c951a05214154a1e6bfb43c728aa714084cea7/armi/operators/settingsValidation.py#L722-L733